src/scheduler/workers/main.ts

```diff
@@ -0,0 +1,264 @@
+import { MilestoneTillAnnualReview } from "../batch/helpers";
+import { addDays, subDays } from "date-fns";
+import { createAnnualReviewProviderUrl, formatDate, isEmailSentBefore, parseDate } from "../helpers";
+import { SCHEDULED_PROCESS_TODAY_DATE } from "server/config";
+
+async function processPostEmailsForList(
+  list: List,
+  milestoneTillAnnualReview: MilestoneTillAnnualReview,
+  reminderType: ListAnnualReviewPostReminderType | ListItemAnnualReviewProviderReminderType
+) {
+  // Check if sent before
+  let emailSent = false;
+  if (list.jsonData.users) {
+    for (const publisherEmail of list.jsonData.users) {
+      const { result } = await sendAnnualReviewPostEmail(
+        milestoneTillAnnualReview,
+        publisherEmail,
+        lowerCase(startCase(list.type)),
+        list?.country?.name ?? "",
+        formatDate(list.nextAnnualReviewStartDate)
+      );
+      if (!emailSent && result) {
+        emailSent = result;
+      }
+    }
+  }
+  // @todo the following code would be used if using Promise.allSettled
+  // const sendResult = await Promise.allSettled(postEmailPromises);
+  // const emailSent = sendResult.find((result) => result.status === "fulfilled" && result.value);
+
+  if (!emailSent) {
+    logger.error(
+      `Unable to send annual review email to post contacts ${list.jsonData.publishers} for list ${list.id} ${milestoneTillAnnualReview} before annual review start`
+    );
+    return;
+  }
+  logger.info(`Annual review email sent to post contacts ${list.jsonData.publishers}`);
+
+  await recordListItemEvent(
+    {
+      eventName: "reminder",
+      itemId: list.id,
+      annualReviewRef: list.jsonData.currentAnnualReview?.reference,
+      reminderType,
+    },
+    AuditEvent.REMINDER,
+    "list"
+  );
+}
+
+async function emailProvider(list: List, listItem: ListItemWithHistory, deletionDate: string) {
+  // @TODO update correct landing page once latest annual review changes are merged in
+  const annualReviewProviderUrl = createAnnualReviewProviderUrl(listItem);
+
+  const providerEmailResult = await sendAnnualReviewProviderEmail(
+    (listItem.jsonData as BaseDeserialisedWebhookData).emailAddress,
+    lowerCase(startCase(listItem.type)),
+    list?.country?.name ?? "",
+    (listItem.jsonData as BaseDeserialisedWebhookData).contactName,
+    deletionDate,
+    annualReviewProviderUrl
+  );
+  return providerEmailResult;
+}
+
+async function processProviderEmailsForListItems(
+  list: List,
+  listItems: ListItemWithHistory[],
+  daysBeforeAnnualReviewStart: string,
+  deletionDate: string
+) {
+  for (const listItem of listItems) {
+    const annualReviewRef = list.jsonData.currentAnnualReview?.reference;
+    let isEmailSent = false;
+
+    // get the most recent audit record to determine if the email has already been sent for the start milestone
+    if (annualReviewRef) {
+      const { result: events } = await findAuditEvents(annualReviewRef, "REMINDER", "listItem");
```

Comment:
you can change findAuditEvents to do something like 

```
prisma.audit.findMany({
    take: 1, //return only 1.. might be 0 based though, soz
    orderBy: {
      time: "desc", // most recent at the "top"
    },
    where: { // that fulfils this criteria
      jsonData: {
        //..
      }
    },
  });
```

instead of pop and loading too much data.